### PR TITLE
Fix teams spawned by trigger action 7,80,107 cannot use IFV or opentopped logic

### LINKED ISSUE #778: Team spawned by trigger action 7,80,107 cannot use IFV or opentopped logic
### First check

- [X] The issue is not introduced by Phobos or any other engine extension, such as Ares, Kratos etc.
- [X] The issue wasn't fixed in the most recent version of Ares/Phobos yet.
- [X] I agree to elaborate the details if requested and provide thorough testing if the bugfix is implemented.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search and read the issue list to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, gifs, videos, debug and exception logs, etc)*.

### Description

This is a known issue, the passengers in the transport vehicle can neither use IFV mode nor fire in open-topped transport
The reason could be the method TeamTypeClass::CreateMembers was implemented badly, a patch or even an entire rework is perhaps needed.
I suggest rewriting this function 

### Conditions to reproduce

none

### INI code

_No response_

### Steps to reproduce

Use trigger action 7, 80 or 107 to spawn a team with an IFV and an infantry / a BFRT and some infantries

### Expected behaviour

IFV modes toggled on,
 passengers can fire normally inside the transport

### Actual behaviour

The transport assumes having no passenger

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chaserli**
Aims to fix #778 
InitialPayload logic won't work.
Needs testing.

### CODE REVIEW

**Comment by chaserli** [comment]
`InitialPayload` is not supported yet, it needs to call Ares' `TechnoExt::ExtData::CreateInitialPayload`, for Ares3.0p1 the address should be  0x446F0

File: src/Ext/Team/Hooks.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+	pPayload->SetLocation(pTransport->Location);
+	pTransport->AddPassenger(pPayload); // ReceiveGunner is done inside FootClass::AddPassenger
+	// Ares' CreateInitialPayload doesn't work here
+	return 0x65DF8D;
+}
+
+DEFINE_HOOK(0x4DE652, FootClass_AddPassenger_NumPassengerGeq0, 0x7)
+{
+	return R->ESI<FootClass*>()->Passengers.NumPassengers > 0 ? 0x4DE65B : 0x4DE666;
```
Comment (Metadorius) [inline src/Ext/Team/Hooks.cpp:31]: would need an enum to denote what different return addresses mean

File: docs/AI-Scripting-and-Mapping.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ This page describes all AI scripting and mapping related additions and changes i
 [Country House]
 RepairBaseNodes=no,no,no ; 3 booleans indicating whether AI repair basenodes in Easy/ Normal/ Difficult game diffculty.
 ```
-
+- Teams spawned by trigger action 7,80,107 can use IFV and opentopped logic normally. `InitialPayload` logic from Ares is not supported yet.
```
Comment (Metadorius) [inline docs/AI-Scripting-and-Mapping.md:17]:
```suggestion
- Teams spawned by trigger action 7,80,107 can use IFV and opentopped logic normally.
  - `InitialPayload` logic from Ares is not supported yet.
```

File: docs/Whats-New.md [hunk 3]
```diff
@@ -257,6 +257,7 @@ New:
 
 Vanilla fixes:
 - Allow AI to repair structures built from base nodes/trigger action 125/SW delivery in single player missions (by Trsdy)
+- Teams spawned by trigger action 7,80,107 can use IFV and opentopped logic normally. (by Trsdy)
```
Comment (Metadorius) [inline docs/Whats-New.md:260]:
```suggestion
- Teams spawned by trigger action 7,80,107 can use IFV and opentopped logic normally (by Trsdy)
```

**Review by Metadorius — APPROVED** [review]

**Pull request merged**